src/errors.rs

```diff
@@ -103,6 +114,12 @@ pub enum IllFormedError {
 impl fmt::Display for IllFormedError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
+            Self::MissedVersion(None) => {
```

Comment:
IMO, `DeclWithoutVersion` makes more sense as the name.  `MissedVersion` doesn't provide very much context.

`MissedEnd` is a little better, enough to make it OK in my opinion, but on reflection maybe it would be better to be more explicit e.g. `MissingEndTag`.

Reply:
I would like to keep errors with a similar meaning named similar. What about
- `MissingDeclVersion`
- `MissingDoctypeName`
- `MissingEndTag`
- `MismatchedEndTag`
- `UnmatchedEndTag`
- `DoubleHyphenInComment` (_not changed_)

?

Reply:
`MissingDeclVersion` is fine, I have no issue with those names.